compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -1627,33 +1628,18 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         };
         let bounds = tcx.item_bounds(def_id).instantiate(tcx, args);
 
-        // The bounds returned by `item_bounds` may contain duplicates after
-        // normalization, so try to deduplicate when possible to avoid
-        // unnecessary ambiguity.
-        let mut distinct_normalized_bounds = FxHashSet::default();
-
         bounds
             .iter()
             .enumerate()
             .filter_map(|(idx, bound)| {
                 let bound_predicate = bound.kind();
                 if let ty::ClauseKind::Trait(pred) = bound_predicate.skip_binder() {
                     let bound = bound_predicate.rebind(pred.trait_ref);
-                    if self.infcx.probe(|_| {
-                        match self.match_normalize_trait_ref(
-                            obligation,
-                            bound,
-                            placeholder_trait_predicate.trait_ref,
-                        ) {
-                            Ok(None) => true,
-                            Ok(Some(normalized_trait))
-                                if distinct_normalized_bounds.insert(normalized_trait) =>
-                            {
-                                true
-                            }
-                            _ => false,
-                        }
-                    }) {
+                    if self.matches_trait_ref(
+                        obligation,
+                        placeholder_trait_predicate.trait_ref,
+                        bound,
+                    ) {
```

Comment:
This shouldn't result in significant changes as winnowing simply prefers earlier projection bounds if they don't guide type inference: https://github.com/rust-lang/rust/blob/5278506a71e65a1c3678e4ea31893eef5820086b/compiler/rustc_trait_selection/src/traits/select/mod.rs#L1918-L1923

It may theoretically weaken type inference, don't think we should care about this however.